compiler/rustc_expand/src/mbe/macro_rules.rs

```diff
@@ -296,6 +299,8 @@ fn generic_extension<'cx, 'tt>(
                 let mut p = Parser::new(sess, tts, false, None);
                 p.last_type_ascription = cx.current_expansion.prior_type_ascription;
 
+                cx.resolver.record_macro_rule_usage(id, i);
```

Comment:
Only need to record use if `id != DUMMY_NODE_ID`, otherwise it's a macro decoded from some other crate.